File: Algorithms/JS/TwoPointers/max-consecutive-ones.js
let nums = [0]

var findMaxConsecutiveOnes = function(nums) {
  let maxCount = 0
  let l = 0
  let r = 0
  while(l < nums.length){
    if (nums[r] == 1){
      r++
    } else{
      maxCount = Math.max(maxCount,r-l)
      r++
      l=r
    }
  }
  return maxCount
}

console.log(findMaxConsecutiveOnes(nums))

File: Java/DSA/src/MaxConsecutiveOnes.java
public class MaxConsecutiveOnes {
    public static void main(String[] args) {
        int[] nums = {1,1,0};
        int count =0;
        int max = 0;
        for(int i=0;i<nums.length;i++){
            if(nums[i] == 1){
                count++;
            }else{
                max = Math.max(max, count);
                count=0;
            }
        }
        max = Math.max(max, count);
        System.out.println(max);
    }
}